docs/UserGuide.md: Add Deadline Field (#101)
AY2425S1-CS2103T-F14A-3/tp: docs/UserGuide.md

### Add Client Details: `add`

> Allows the user to add a new client with various details including personal information, payment status, client status, and project status.

Reviewer: ```suggestion
Allows the user to add a new client with details including client name and contact details, payment status, client status, project status and project deadline.
```

Format: `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS d/DEADLINE [t/TAG]…​
[ps/PROJECT_STATUS] [py/PAYMENT_STATUS] [cs/CLIENT_STATUS]`

* Clients with the **same** `NAME`, `EMAIL` and `PHONE NUMBER` are considered duplicates and will not be added
* A person can have any number of tags (including 0)
* `NAME` must be **alphanumeric**, may contain **spaces** and **dashes**, and should not be blank.

> * `NAME` is case-insensitive. `John Doe` and `joHN dOE` are considered same clients, but name is stored in the same case as the input so `John Doe` is stored as `John Doe` and `JOHN Doe` is stored as `JOHN Doe`

Reviewer: ```suggestion
* `NAME` is case-insensitive. `John Doe` and `joHN dOE` are considered same clients, but name is stored in the same case as the input (so `John Doe` is stored as `John Doe` and `JOHN Doe` is stored as `JOHN Doe`)
```

* `NAME` must not exist in Clientele+ already.
* `PHONE_NUMBER` should be **Numeric** digits,may include “-” or spaces. Example: `555-1234` or `555 1234`.
* `EMAIL`  Standard email format “user@example.com”
* `PAYMENT STATUS` Acceptable values are `pending`, `partial`, `paid`, `late`. Case insensitive.
* `CLIENT STATUS`  Acceptable values are `active`, `unresponsive`, `potential`, `old`. Case sensitive.
* `PROJECT STATUS` Acceptable values are `in progress`, `completed`. Case insensitive.
* `TAG` Should be alphanumeric. May contain spaces.

> * `DEADLINE` Should be in the following format: `dd-MM-yyyy`. Example: `10-10-2024` represents **10th October 2024**

Reviewer: ```suggestion
* `DEADLINE` Should be in the following format: `DD-MM-YYYY`. Example: `10-10-2024` represents **10th October 2024**
```

Reviewer: Or make it `dd-mm-yyyy` (i.e. all uppercase or all lowercase).
